at_client/lib/src/service/notification_service_impl.dart

```diff
@@ -0,0 +1,85 @@
+import 'dart:convert';
+import 'package:at_client/at_client.dart';
+import 'package:at_client/src/manager/monitor.dart';
+import 'package:at_client/src/preference/monitor_preference.dart';
+import 'package:at_client/src/service/notification_service.dart';
+import 'package:at_commons/at_commons.dart';
+
+class NotificationServiceImpl implements NotificationService {
+  Map<String, NotificationService> instances = {};
+  Map<String, Function> listeners = {};
+  final EMPTY_REGEX = '';
+  static const notificationIdKey = '_latestNotificationId';
+
+  late AtClient atClient;
+
+  NotificationService? getInstance() {
+    _startMonitor();
+    return instances[atClient.getCurrentAtSign()];
+  }
+
+  void _startMonitor() async {
+    final lastNotificationId = await _getLastNotificationId();
+    final monitor = Monitor(
```

Comment:
We are using Monitor directly. Is there a need for MonitorService class?

Reply:
No..MonitorService is replaced with NotificationService in new code.